### PR TITLE
Replace JTF.RunAsync() with exception-safe wrappers

### PR DESCRIPTION
**Pull request by mstv**
Follow-up to #11136 (and https://github.com/gitextensions/gitextensions/pull/11137#discussion_r1293391852)
inspired by d5b8f33de75f04606948ccd632259b9d388eb234

## Proposed changes

- Replace `JoinableTaskFactory.RunAsync()` with exception-safe wrappers `FileAndForget` or `InvokeAndForget`, respectively
- Replace and remove `GitUIExtensions.InvokeSync()` and `GitUIExtensions.InvokeAsync()`

## Screenshots 

N/A

## Test methodology 

- exisiting tests

## Please do not squash merge ❗

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitExtUtils/GitUI/TaskManager.cs
Comment (RussKie) [inline GitExtUtils/GitUI/TaskManager.cs:149]: Write the exception to `Trace` stream so that we can observe it even outside a debugger?

File: GitUI/CommandsDialogs/FormCommit.cs
Comment (mstv) [reply GitUI/CommandsDialogs/FormCommit.cs:1189]: The injection of a JTF instance should be no big deal: The class `TaskManager` provides / encapsulates the JTF instance.
With the initial implementation, it would look like:
```
_taskManager.InvokeAndForget(SelectedDiff, () => SelectedDiff.ViewChangesAsync(...));
```
With another argument added to the extension method, it can be written as before (preferred):
```
SelectedDiff.InvokeAndForget(() => SelectedDiff.ViewChangesAsync(...), _taskManager);
```

---

Furthermore, I should feed every already available `CancellationToken` to `InvokeAndForget()`, too - but in another follow-up PR:

```
CancellationToken cancellationToken = _viewChangesSequence.Next();
SelectedDiff.InvokeAndForget(() => SelectedDiff.ViewChangesAsync(..., cancellationToken), _taskManager, cancellationToken);
```

File: GitExtUtils/GitUI/TaskManager.cs
Comment (mstv) [reply GitExtUtils/GitUI/TaskManager.cs]: done

**Review by RussKie — APPROVED** [review]

**Comment by mstv** [comment]
On hold: The test execution will improve / work again after merging the 3 today's PRs.

**Comment by mstv** [comment]
needs #11162

**Review by RussKie — APPROVED** [review]

**Pull request merged**
